traits/constants.py

```diff
@@ -215,4 +219,5 @@ class DefaultValue(IntEnum):
     DefaultValue.callable_and_args: "factory",
     DefaultValue.callable: "method",
     DefaultValue.trait_set_object: "set",
+    DefaultValue.disallow: "invalid",
```

Comment:
```suggestion
    DefaultValue.disallow: "disallow",
```

Reply:
Yep, the choice not to use "disallow" here was deliberate. In as much as the `default_kind` is used for anything, it's a description of the type of the default value (`"value"`, `"set"`, `"method"`, etc.), and `"invalid"` seemed a better fit than `"disallow"` here.

Reply:
(Bigger picture: one of my goals with all this refactoring is to make the default value handling consistent and clean enough that the `default_kind` property is clearly not useful.)

Reply:
We could always rename from "disallow" to "invalid" throughout, if anyone feels it's important enough.

Reply:
ahh. sounds good to me. resolving.